horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/Util.java

```diff
@@ -559,26 +593,87 @@ static <T> void evaluateWithCombinationFunction(List<T> inputData,
       }
    }
 
+   private static Context createContext(OutputStream out){
+      return Context.newBuilder("js")
+              .engine(Engine.newBuilder()
+                      .option("engine.WarnInterpreterOnly", "false")
+                      .build()
+              )
+              .allowExperimentalOptions(true)
+              .option("js.foreign-object-prototype", "true")
+              .option("js.global-property","true")
+              .out(out)
+              .err(out)
+              .build();
+   }
+   private static void setupContext(Context context) throws IOException {
+      context.getBindings("js").putMember("isInstanceLike", new ProxyJacksonObject.InstanceCheck());
+      context.eval("js",
+              "Object.defineProperty(Object,Symbol.hasInstance, {\n" +
+                      "  value: function myinstanceof(obj) {\n" +
+                      "    return isInstanceLike(obj);\n" +
+                      "  }\n" +
+                      "});");
+   }
+
+   public static Value resolvePromise(Value value){
+      if(value.getMetaObject().getMetaSimpleName().equals("Promise") && value.hasMember("then") && value.canInvokeMember("then")){
+         List<Value> resolved = new ArrayList<>();
+         List<Value> rejected = new ArrayList<>();
+         Object invokeRtrn = value.invokeMember("then", new ProxyExecutable() {
+            @Override
+            public Object execute(Value... arguments) {
+               resolved.addAll(Arrays.asList(arguments));
+               return arguments;
+            }
+         }, new ProxyExecutable() {
+            @Override
+            public Object execute(Value... arguments) {
+               rejected.addAll(Arrays.asList(arguments));
+               return arguments;
+            }
+         });
+         if(invokeRtrn instanceof Value){
+            Value invokedValue = (Value)invokeRtrn;
+         }
+         if(!rejected.isEmpty()){
+            value = rejected.get(0);
+         }else if(resolved.size() == 1){
+            value = resolved.get(0);
+         }else{ //resolve.size() > 1, this doesn't happen
+
+         }
```

Comment:
Remove empty `else` stmt (we could still keep the comment `//resolve.size() > 1 doesn't happen` if we want)
```suggestion
         }
```

Reply:
instead of an empty `else` perhaps we should log an error that the resolve violated Horreum's expectation?

Reply:
yeah makes sense, so if it happens we have logs to track it

Reply:
I will update it to include a `log.error` rather than just a comment stating the else branch is unexpected.